### PR TITLE
[opichuryn] sequence_task_1

### PR DESCRIPTION
**Pull request by opichuryn**

### CODE REVIEW

File: sequence_task_1/student/include/impl.h [hunk 1]
```diff
@@ -7,34 +7,86 @@ template<class Container, class Song_t>
...
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
+    using self_type = StaticPlaylist<Container, Song_t>;
 
-    /** @todo Iterators */
+    /** Iterators */
+    const_iterator begin() {
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h:21]: why not marked as const method?

File: sequence_task_1/student/include/impl.h [hunk 2]
```diff
@@ -7,34 +7,86 @@ template<class Container, class Song_t>
...
     StaticPlaylist() = default;
+    StaticPlaylist(const self_type& other) = default;
+    StaticPlaylist(self_type&& other) = default;
+    self_type& operator=(const self_type& rhs) = default;
+    self_type& operator=(self_type&& rhs) = default;
 
-    /** @todo Constructor from any reversible sequence container */
+    /** Constructor from any reversible sequence container */
+    template <class AnyContainer>
+    StaticPlaylist(AnyContainer&& container)
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h:37]: why not const?

File: sequence_task_1/student/include/impl.h (hunk 1 above)
Comment (opichuryn) [reply sequence_task_1/student/include/impl.h]: Done!

File: sequence_task_1/student/include/impl.h (hunk 2 above)
Comment (opichuryn) [reply sequence_task_1/student/include/impl.h]: Done!

**Pull request closed without merging**